lib/ssg/downloader.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require 'json'
+require 'net/http'
+
+module Ssg
+  # Downloads SCAP datastreams from the SCAP Security Guide
+  # https://github.com/ComplianceAsCode/content
+  class Downloader
+    LATEST_RELEASE_URI = URI('https://api.github.com/repos'\
+                             '/ComplianceAsCode/content/releases/latest')
+    SSG_DS_REGEX = /scap-security-guide-(\d+\.)+zip$/
+
+    def self.download!
+      new.fetch_datastream_file
+    end
+
+    def fetch_datastream_file
+      get_chunked(datastream_uri)
+
+      datastream_filename
+    end
+
+    private
+
+    def datastream_uri
+      @datastream_uri ||= URI(
+        download_urls.find { |url| url[SSG_DS_REGEX] }
+      )
+    end
+
+    def download_urls
+      get_json(LATEST_RELEASE_URI).dig('assets').map do |asset|
+        asset.dig('browser_download_url')
```

Comment:
This method is coupled to GitHub's releases api, unfortunately. If they change the structure of the response, we'll have to update this method.